aleph-client/src/rpc.rs

```diff
@@ -10,18 +11,95 @@ fn json_req(method: &str, params: Value, id: u32) -> Value {
     })
 }
 
-pub fn author_rotate_keys() -> Value {
+pub fn author_rotate_keys_json() -> Value {
     json_req("author_rotateKeys", Value::Null, 1)
 }
 
+fn state_query_storage_at_json(storage_keys: &Vec<StorageKey>) -> Value {
+    json_req(
+        "state_queryStorageAt",
+        Value::Array(vec![
+            Value::Array(
+                storage_keys
+                    .into_iter()
+                    .map(|storage_key| Value::String(hex::encode(storage_key)))
+                    .collect::<Vec<_>>(),
+            ),
+            Value::Null,
+        ]),
+        1,
+    )
+}
+
+fn parse_query_storage_at_result(
+    maybe_json_result: Option<String>,
+) -> Result<Vec<Option<String>>, String> {
```

Comment:
from what I saw we expect this Vec to be full of `Somes` and if it has None it means something went wrong. Lets just return here Result<Vec<String>, String> and treat the None as an error already here

Reply:
You're right but this is really up to the user of `parse_query_storage_at`. I mean that `null` returned for a given storage key from `parse_query_storage_at` it's not a fatal error. So far we have one user of `parse_query_storage_at` which is `locks`, and that function treats `None` as error. One can add at some point different function that depends on generic `parse_query_storage_at` and I did not want to close that door.